api/client/src/main/java/org/projectnessie/client/http/impl/jdk11/JavaHttpClient.java

```diff
@@ -75,6 +77,26 @@ public URI getBaseUri() {
 
   @Override
   public void close() {
+    if (client != null) {
+      client
+          .executor()
+          .ifPresent(
+              executor -> {
+                if (executor instanceof ExecutorService) {
+                  ExecutorService executorService = (ExecutorService) executor;
+                  executorService.shutdown();
+                  try {
+                    if (!executorService.awaitTermination(10, TimeUnit.SECONDS)) {
+                      executorService.shutdownNow();
+                    }
+                  } catch (InterruptedException e) {
+                    executorService.shutdownNow();
```

Comment:
just FYI if this is supposed to mimic the "default impl" of guava's `MoreExecutors.shutdownAndAwaitTermination` (which we cant use in the client code i guess)
we are getting some details wrong...

also wondering if the 10 sec timeout should instead be related to our http client default timeout (if we have that, didnt check :sweat_smile:)

Reply:
a. Yes, I am aware of `MoreExecutors.shutdownAndAwaitTermination`, but this module doesn't depend on those lib. Hence, I didn't use it.

b. do you mean the order of executorService.shutdownNow(); and Thread.currentThread().interrupt();?

c. Used 10 seconds because of existing usages.
https://github.com/projectnessie/nessie/blob/627e9d73144b346c8ce5e8c16497a9a61bc0b8b5/api/client/src/main/java/org/projectnessie/client/auth/oauth2/OAuth2TokenRefreshExecutor.java#L40-L50

Reply:
My implementation was based on oracle docs `shutdownAndAwaitTermination`

https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/util/concurrent/ExecutorService.html

Reply:
afaik both your link and guava do some more waiting after calling `shutdownNow` but the code in this PR does not

Reply:
if we have multiple places in the client code we might want to have a single re-usable helper method potentially.